packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/PushNotificationBackgroundService.kt

```diff
@@ -0,0 +1,89 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazonaws.amplify.amplify_push_notifications
+
+import android.content.Context
+import android.content.Intent
+import android.os.Handler
+import android.util.Log
+import androidx.core.app.JobIntentService
+import com.amplifyframework.pushnotifications.pinpoint.utils.processRemoteMessage
+import com.google.firebase.messaging.RemoteMessage
+import io.flutter.embedding.engine.FlutterEngine
+import io.flutter.embedding.engine.dart.DartExecutor
+import io.flutter.embedding.engine.loader.FlutterLoader
+import io.flutter.view.FlutterCallbackInformation
+import java.util.*
+
+private const val TAG = "PushBackgroundService"
+
+class PushNotificationBackgroundService : JobIntentService() {
+
+    /**
+     * The Background Flutter engine that initializes the callback dispatcher
+     */
+    private var backgroundFlutterEngine: FlutterEngine? = null
```

Comment:
Could `backgroundFlutterEngine` be `lateinit` non-nullable? And use another flag to check if the background service has started in `startAndHandlePushNotificationService`?

Reply:
Keeping this nullable simplifies the caching logic later. I still introduced another flag that was needed.